macOS/DuckDuckGo/CrashReports/Model/CrashReportReader.swift

```diff
@@ -17,68 +17,99 @@
 
 import Foundation
+import Common
 
 final class CrashReportReader {
 
-    static let displayName = Bundle.main.displayName
     static let vpnExtensionDisplayName = "com.duckduckgo.macos.vpn.network-extension"
 
+    static func validBundleIdentifiers() -> [String] {
+        return [
+            Bundle.main.bundleIdentifier,
+            Bundle.main.vpnMenuAgentBundleId,
+            Bundle.main.vpnSystemExtensionBundleId,
+            Bundle.main.vpnProxyExtensionBundleId,
+            Bundle.main.dbpBackgroundAgentBundleId
+        ].compactMap(\.self)
+    }
+
+    private let fileManager: FileManager
+    private let validBundleIdentifierProvider: () -> [String]
+    private let dateProvider: () -> Date
+
+    init(fileManager: FileManager = .default,
+         validBundleIdentifierProvider: @escaping () -> [String] = CrashReportReader.validBundleIdentifiers,
+         dateProvider: @escaping () -> Date = Date.init) {
+        self.fileManager = fileManager
+        self.validBundleIdentifierProvider = validBundleIdentifierProvider
+        self.dateProvider = dateProvider
+    }
+
     func getCrashReports(since lastCheckDate: Date) -> [CrashReport] {
         var allPaths: [URL]
 
         do {
-            allPaths = try FileManager.default.contentsOfDirectory(at: FileManager.userDiagnosticReports, includingPropertiesForKeys: nil)
+            allPaths = try fileManager.contentsOfDirectory(at: FileManager.userDiagnosticReports, includingPropertiesForKeys: nil)
         } catch {
             assertionFailure("CrashReportReader: Can't read content of diagnostic reports \(error.localizedDescription)")
             return []
         }
 
         do {
-            let systemPaths = try FileManager.default.contentsOfDirectory(at: FileManager.systemDiagnosticReports, includingPropertiesForKeys: nil)
+            let systemPaths = try fileManager.contentsOfDirectory(at: FileManager.systemDiagnosticReports, includingPropertiesForKeys: nil)
             allPaths.append(contentsOf: systemPaths)
         } catch {
             assertionFailure("Failed to read system crash reports: \(error)")
         }
 
         let filteredPaths = allPaths.filter({
-            isCrashReportPath($0) && belongsToThisApp($0) && isFile(at: $0, newerThan: lastCheckDate)
+            isCrashReportPath($0) && isFile(at: $0, newerThan: lastCheckDate)
         })
 
-        return filteredPaths.compactMap(crashReport(from:))
+        return filteredPaths
+            .compactMap(crashReport(from:))
+            .filter(matchesBundleID)
     }
 
     private func isCrashReportPath(_ path: URL) -> Bool {
         let validExtensions = [LegacyCrashReport.fileExtension, JSONCrashReport.fileExtension]
-        return validExtensions.contains(path.pathExtension)
+        guard validExtensions.contains(path.pathExtension) else {
+            return false
+        }
+
+        let fileName = path.lastPathComponent.lowercased()
+        return fileName.contains("duckduckgo")
```

Comment:
We now do this in the following steps:
1. Filter crash report URLs by those that have `duckduckgo` in them, to better capture cases where we have a bundle ID as the product name, like the proxy or VPN extension.
2. Parse the contents of filtered URLs into `CrashReport` objects
3. Check that the bundle IDs on the `CrashReport` objects are included in `validBundleIdentifiers`, which will cause those from build variants like review to get filtered out

Reply:
This sounds great to me 👏